libs/ai-endpoints/langchain_nvidia_ai_endpoints/_common.py

```diff
@@ -368,13 +368,17 @@ def _aggregate_msgs(self, msg_list: Sequence[dict]) -> Tuple[dict, bool]:
         content_buffer: Dict[str, Any] = dict()
         content_holder: Dict[Any, Any] = dict()
         usage_holder: Dict[Any, Any] = dict()  ####
+        finish_reason_holder: Optional[str] = None
         is_stopped = False
         for msg in msg_list:
             usage_holder = msg.get("usage", {})  ####
             if "choices" in msg:
                 ## Tease out ['choices'][0]...['delta'/'message']
                 msg = msg.get("choices", [{}])[0]
-                is_stopped = msg.get("finish_reason", "") == "stop"
+                # todo: this meeds to be fixed, the fact we only
+                #       use the first choice breaks the interface
+                finish_reason_holder = msg.get("finish_reason", None)
```

Comment:
I guess this variable is not necessary: finish_reason_holder

is_stopped = msg.get("finish_reason") == "stop" should do?
and the if finish_reason_holder: can use is_stopped?

Reply:
it's also used in https://github.com/langchain-ai/langchain-nvidia/pull/72/files#diff-6b48e6b0775169741d8200a861c8bbab23256e2d1986caa9cd56d74624993ee4R399